src/modules/musig/session_impl.h

```diff
@@ -670,6 +670,43 @@ int secp256k1_musig_partial_sig_verify(const secp256k1_context* ctx, const secp2
      * negated exactly when the aggregate key parity is odd. If the aggregate
      * key is tweaked, then negation happens when the aggregate key has an odd Y
      * coordinate XOR the internal key has an odd Y coordinate.*/
+
+    /* When producing a partial signature, signer i uses a possibly
+     * negated secret key:
+     *
+     *   sk[i] = (d_tweak*d_agg*d[i])*x[i]
+     *
+     * to ensure that the aggregate signature will correspond to
+     * an aggregate public key with even Y coordinate (see the
+     * notation and explanation in musig_partial_sign).
+     *
+     * We use the following additional notation:
+     * - e is the (Schnorr signature) challenge
+     * - r[i] is the i-th signer's secret nonce
+     * - R[i] = r[i]*G is the i-th signer's public nonce
+     * - R is the aggregated public nonce
+     * - d_nonce is chosen so that |R| = d_nonce*R
+     *
+     * The i-th partial signature is:
+     *
+     *   s[i] = d_nonce*r[i] + mu[i]*e*sk[i]
+     *
+     * In order to verify this partial signature, we need to check:
+     *
+     *   s[i]*G = d_nonce*R[i] + mu[i]*e*sk[i]*G
+     *
+     * The verifier doesn't have access to sk[i]*G, but can construct
+     * it using the xonly public key |P[i]| as follows:
+     *
+     *   sk[i]*G = d_tweak*d_agg*d[i]*x[i]*G
+     *           = d_tweak*d_agg*d[i]*P[i]
+     *           = d_tweak*d_agg*|P[i]|
+     *
+     * The if condition is below is true whenever d_tweak*d_agg is
```

Comment:
Oops, sorry, typo introduced in previous suggestion

```suggestion
     * The if condition below is true whenever d_tweak*d_agg is
```